signer/src/api/new_block.rs

```diff
@@ -112,34 +143,266 @@ pub async fn new_block_handler(state: State<ApiState<impl Context>>, body: Strin
-        tracing::warn!(%result, "Failed to set chainstate in emily");
+    let futures = vec![
+        emily_client
+            .update_deposits(completed_deposits)
+            .map(UpdateResult::Deposit)
+            .boxed(),
+        emily_client
+            .update_withdrawals(updated_withdrawals)
+            .map(UpdateResult::Withdrawal)
+            .boxed(),
+        emily_client
+            .create_withdrawals(created_withdrawals)
+            .map(UpdateResult::CreatedWithdrawal)
+            .boxed(),
+        emily_client
+            .set_chainstate(chainstate)
+            .map(UpdateResult::Chainstate)
+            .boxed(),
+    ];
+    // TODO: Ideally, we would use `futures::future::join_all` here, but Emily
+    // randomly returns a `VersionConflict` error when we send multiple
+    // requests that may update the chainstate.
+    // let results = futures::future::join_all(futures).await;
+
+    // Log any errors that occurred while updating Emily.
+    // We don't return a non-success status code here because we rely on
+    // the redundancy of the other sBTC signers to ensure that the update
+    // is sent to Emily.
+    for future in futures {
+        let result = future.await;
+        if let UpdateResult::Chainstate(Err(error)) = result {
+            tracing::warn!(%error, "Failed to set chainstate in emily");
+        } else if let UpdateResult::Deposit(Err(error)) = result {
+            tracing::warn!(%error, "Failed to update deposits in emily");
+        } else if let UpdateResult::Withdrawal(Err(error)) = result {
+            tracing::warn!(%error, "Failed to update withdrawals in emily");
+        } else if let UpdateResult::CreatedWithdrawal(results) = result {
+            for result in results {
+                if let Err(error) = result {
+                    tracing::warn!(%error, "Failed to create withdrawals in emily");
+                }
+            }
+        }
     }
-
     StatusCode::OK
 }
 
+/// Fetches Bitcoin block information based on a transaction ID from the database.
+///
+/// # Parameters
+/// - `txid`: The transaction ID used to look up the Bitcoin block data.
+/// - `db`: Database access to retrieve the Bitcoin block information.
+///
+/// # Returns
+/// Returns a `Result` which:
+/// - On success, contains the `BitcoinBlock` information.
+/// - On failure, returns an `Error`, either because no block was found for the provided transaction ID
+///   or due to an error in database access.
+///
+/// # Notes
+/// - If the transaction is missing from any blocks, it returns an `Error::BitcoinTxMissing`.
+/// - In case of database access errors, the caller should handle the error appropriately.
+async fn fetch_bitcoin_block_from_txid(
+    db: &impl DbRead,
+    txid: Txid,
+) -> Result<BitcoinBlock, Error> {
+    let btc_txid = BitcoinTxId::from(txid);
+    let blocks_hashes = db.get_bitcoin_blocks_with_transaction(&btc_txid).await?;
```

Comment:
The necessary bitcoin block info is given in the webhook body. Better to use the webhook body exclusively because our database might not be up to date by the time this function runs.

Reply:
I have a question regarding this. From my understanding, the body of the webhook contains information about the Bitcoin tip (and its parent). The `Fulfillment` struct requires `The Bitcoin block hash of the block that included the fulfilling Bitcoin transaction` (from the docs). Can we be certain that the Bitcoin block information in the webhook corresponds to the block fulfilling the transaction, or could it potentially be more recent?

Reply:
> The `Fulfillment` struct requires `The Bitcoin block hash of the block that included the fulfilling Bitcoin transaction` (from the docs). Can we be certain that the Bitcoin block information in the webhook corresponds to the block fulfilling the transaction, or could it potentially be more recent?

Oh right. To do this properly we need to wrap up https://github.com/stacks-network/sbtc/pull/493. For now I'd add a todo comment there about the dependency and put some dummy value (like the current bitcoin chain tip or something).